### ISSUE TITLE
Carbs recorded when Adding a Carb Entry is blocked because a bolus is in progress

### ISSUE THREAD

**Issue by alfredoig** [LoopKit/Loop#2202]
**Describe the bug**
Carbs recorded when Adding a Carb Entry bolus is blocked because a bolus is in progress.

**Attach an Issue Report**
~~Tap the Loop settings icon on the bottom of the screen, then tap Issue Report and attach it to this ticket.~~

**To Reproduce**
Steps to reproduce:
1. Have a bolus in progress (make sure it is still in progress when step 6 is executed)
<img width="385" alt="image" src="https://github.com/user-attachments/assets/f979939b-818c-43da-abf0-041e8efb08d0">

2. Click on Add Carb Entry
3. Enter Carbs
4. Click Continue
5. Make sure a bolus is required on Meal Bolus Screen (or enter an amount of insulin to be delivered with the carbs)
<img width="386" alt="image" src="https://github.com/user-attachments/assets/c71f6201-01b1-4a23-88ed-6f8c41083b8b">

6. Save and Deliver
7. Message "Bolus in progress. Wait for existing bolus to finish, or cancel bolus" is displayed.
<img width="393" alt="image" src="https://github.com/user-attachments/assets/51b83d1f-e2f1-4dc1-9414-d07f42e3caa1">

8. The carbs are stored but no insulin delivery is given.
<img width="387" alt="image" src="https://github.com/user-attachments/assets/4b5e6fba-87b3-4c32-a5d4-1274fbfe8fd7">

9. **This cause issues as the insulin delivery was canceled but the carbs are recorded., specially when Dosing Strategy: Automatic Bolus is selected**

**Expected behavior**
If the Meal Bolus is canceled the carbs shouldn't be recorded.

**Screenshots**
On the Steps to reproduce

**Phone**
 - Hardware: iPhone  12 Pro
 - OS Version: 17.6. Version 17.5 also had the same problems

**Loop Version**
 - Version Number: 3.4.1 (57), version 3.2.3 also had the same issue.
 - Repo: LoopKit/Loop

**CGM**
 - Device: Dexcom G6
 - Manager app: Dexcom App

**Pump**
 - Manufacturer: Insulet
 - Model: Omnipod Dash
 - Firmware version: 4.10.0

**Additional context**
~~Add any other context about the problem here.~~

**Comment by elnjensen** [comment]
I think what the "expected behavior" is here may be a matter of opinion.  To me, the current behavior is exactly what I expect, and more convenient than the alternative would be.  

Saving carbs and bolusing, though related, are two distinct things.  If I hit a button to save carbs, I expect them to be saved, even if the bolus can't be given.  I can always hit the bolus button again after waiting for the current bolus to finish, without having to enter the carbs again.  

It may be that the warning message could be clearer that the carbs were saved, even if bolus wasn't delivered.

**Comment by marionbarker** [comment]
This behavior is by design  

https://loopkit.github.io/loopdocs/operation/features/carbs/#meal-entry_1

https://loopkit.github.io/loopdocs/operation/features/carbs/?h=double#avoid-double-meal-entries

please close this issue.

**Comment by alfredoig** [comment]
I raised it as an issue because I am going to eat a meal and I prebolus (10-15 mins), but my glucose is going up and I am getting a bolus when I enter my Meal. The meal bolus is not given as the correction bolus is given.

I have closed the issue, thank you for the information.

**Comment by marionbarker** [comment]
Yes - please close this issue. Note - you can continue adding comments to a closed issue. People watching the repo or people who have previously commented on a closed issue will still get an email.

The questions you are asking are more appropriate for a support forum like zulipchat or facebook.
* https://loopkit.github.io/loopdocs/intro/loopdocs-how-to/#how-to-find-help

The code is acting as designed. You are welcome to disagree with the design - if that is the case, zulipchat is your best place for discussion.

The current meal and bolus interface evolved over years and it is unlikely to be modified in the way you are suggesting. The carbs are saved whether the bolus goes through or not.

Here are some ideas for you to consider - but please - if you have more questions - use one of the support groups not an Issue.

1. If you can't remember to go back and add a bolus after the bolus completes, just tap the Cancel Bolus row and then bolus immediately. Loop only automatically boluses once every 5 minutes.
2. The 30 min lollipop is intended only for low treatments - do not use it for meal entries
3. You can use a meal override where the low end of the range is as low as your glucose suspend threshold - in that case, Loop will auto-bolus (or increase temp basal) sooner than using your normal target range

**Comment by alfredoig** [comment]
Hi @marionbarker,

Thank you for the information, and I understand how the system works. What happened to me it is that I received a failure to deliver message and I assumed the whole transaction (Carbs and Bolus) failed. Then I did the transaction again, as it failed before, to my surprise the Carbs where recorded twice. I investigated what happened and realised the Carbs were recorded but not the Bolus in the first transaction, this was a defect in the software to me and the reason why I submitted it as a defect.

Thank you for the support, I am closing the ticket now.
